MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -58,18 +123,127 @@ public bool ValidateVersion(string encoderAppPath, bool logOutput)
+            FFmpegVersion underTest = GetFFmpegVersion(output);
 
-            for (var i = 2013; i <= 2015; i++)
+            if (logOutput)
             {
-                var yearString = i.ToString(CultureInfo.InvariantCulture);
-                if (output.IndexOf(" " + yearString + " ", StringComparison.OrdinalIgnoreCase) != -1)
+                if (underTest.Unknown())
+                {
+                    _logger.LogWarning("FFmpeg validation: Unknown version");
+                }
+                else if (underTest.Below(required))
                 {
-                    return false;
+                    _logger.LogWarning("FFmpeg validation: Found version {0} which is below the minimum recommended of {1}",
+                        underTest.ToString(), required.ToString());
                 }
+                else if (underTest.Experimental())
+                {
+                    _logger.LogWarning("FFmpeg validation: Unknown version: {0}?", underTest.ToString());
+                }
+                else
+                {
+                    _logger.LogInformation("FFmpeg validation: Detected version {0}", underTest.ToString());
+                }
+            }
+
+            return underTest.Suitable(required);
+        }
+
+        /// <summary>
+        /// Using the output from "ffmpeg -version" work out the FFmpeg version.
+        /// For pre-built binaries the first line should contain a string like "ffmpeg version x.y", which is easy
+        /// to parse.  If this is not available, then we try to match known library versions to FFmpeg versions.
+        /// If that fails then we use one of the main libraries to determine if it's new/older than the latest
+        /// we have stored.
+        /// </summary>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        static private FFmpegVersion GetFFmpegVersion(string output)
+        {
+            // For pre-built binaries the FFmpeg version should be mentioned at the very start of the output
+            var match = Regex.Match(output, @"ffmpeg version (\d+\.\d+)");
+
+            if (match.Success)
+            {
+                return new FFmpegVersion(match.Groups[1].Value);
+            }
+            else
+            {
+                // Try and use the individual library versions to determine a FFmpeg version
+                // This lookup table is to be maintained with the following command line:
+                // $ ./ffmpeg.exe -version | perl -ne ' print "$1=$2.$3," if /^(lib\w+)\s+(\d+)\.\s*(\d+)/'
+                ReadOnlyDictionary<FFmpegVersion, string> lut = new ReadOnlyDictionary<FFmpegVersion, string>
+                    (new Dictionary<FFmpegVersion, string>
+                    {
+                        { new FFmpegVersion("4.1"), "libavutil=56.22,libavcodec=58.35,libavformat=58.20,libavdevice=58.5,libavfilter=7.40,libswscale=5.3,libswresample=3.3,libpostproc=55.3," },
```

Comment:
Create lookup table detailing what lib versions are in each FFmpeg version.  We can then grab just the lib versions from the '-version' output and search the dictionary for match, revealing the FFmpeg version...  How likely is this?